src/uproot/behaviors/TBranch.py

```diff
@@ -3549,6 +3578,187 @@ def delayed_get_array(ttree, key):
+                real_filter_branch = filter_branch
+
+            hasbranches.append(obj)
+
+            new_keys = obj.keys(
+                recursive=recursive,
+                filter_name=filter_name,
+                filter_typename=filter_typename,
+                filter_branch=real_filter_branch,
+                full_paths=full_paths,
+            )
+
+            if common_keys is None:
+                common_keys = new_keys
+            else:
+                new_keys = set(new_keys)
+                common_keys = [key for key in common_keys if key in new_keys]
+
+    if count == 0:
+        raise ValueError(
+            "allow_missing=True and no TTrees found in\n\n    {}".format(
+                "\n    ".join(
+                    "{"
+                    + "{}: {}".format(
+                        repr(f.file_path if isinstance(f, HasBranches) else f),
+                        repr(f.object_path if isinstance(f, HasBranches) else o),
+                    )
+                    + "}"
+                    for f, o in files
+                )
+            )
+        )
+
+    if len(common_keys) == 0 or not (all(is_self) or not any(is_self)):
+        raise ValueError(
+            "TTrees in\n\n    {}\n\nhave no TBranches in common".format(
+                "\n    ".join(
+                    "{"
+                    + "{}: {}".format(
+                        repr(f.file_path if isinstance(f, HasBranches) else f),
+                        repr(f.object_path if isinstance(f, HasBranches) else o),
+                    )
+                    + "}"
+                    for f, o in files
+                )
+            )
+        )
+
+    class _UprootRead:
+        def __init__(self, hasbranches, branches) -> None:
+            self.hasbranches = hasbranches
+            self.branches = branches
+
+        def __call__(self, i_start_stop):
+            i, start, stop = i_start_stop
+            return self.hasbranches[i].arrays(
+                self.branches, entry_start=start, entry_stop=stop
+            )
+
+    partition_args = []
+    for i, ttree in enumerate(hasbranches):
+        entry_start, entry_stop = _regularize_entries_start_stop(
+            ttree.num_entries, None, None
+        )
+        entry_step = 0
+        if uproot._util.isint(step_size):
+            entry_step = step_size
+        else:
+            entry_step = ttree.num_entries_for(step_size, expressions=common_keys)
+
+        def foreach(start):
+            stop = min(start + entry_step, entry_stop)
+            partition_args.append((i, start, stop))
+
+        for start in range(entry_start, entry_stop, entry_step):
+            foreach(start)
+
+    first5 = hasbranches[0].arrays(common_keys, entry_start=0, entry_stop=5)
+    meta = dask_awkward.core.typetracer_array(first5)
+
+    return dask_awkward.from_map(
+        _UprootRead(hasbranches, common_keys),
+        partition_args,
+        label="from-uproot",
+        meta=meta,
+    )
```

Comment:
Relevant dask part cc @douglasdavis

Reply:
Notes: First I tried using the same code to create the typetracer array with `ak.Array(first5.layout.typetracer.forget_length())` but it was throwing an error that made me realise it probably wasn't the correct way to get the tytracer array.

Going through the code for dask-awkward I found this function `dask_awkward.core.typetracer_array`, it works, but please confirm if I'm using it right.

Reply:
What motivated the number 5? Is it possible for these 5 entries to not have something that you're looking for, and in that case, does it fail?

Since reading happens at a per-TBasket level, getting the whole first TBasket might make more sense. [TBranch.basket_entry_start_stop](https://uproot.readthedocs.io/en/latest/uproot.behaviors.TBranch.TBranch.html#basket-entry-start-stop) would give you the `entry_start` and `entry_stop` numbers for a given TBasket (such as number `0`).